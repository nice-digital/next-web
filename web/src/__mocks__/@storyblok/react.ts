export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
