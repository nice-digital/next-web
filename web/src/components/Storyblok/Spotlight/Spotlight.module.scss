@use '@nice-digital/nds-core/scss/layout';
@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';
@use '~/styles/colours';
@use '~/styles/spacing' as *;

.spotlight {
  @include layout.fullWidth();
  background: colours.$spotlight-background;
  padding: $homepage-large-spacing 0;
}

.container {
  @include layout.fullWidthContainer();
}

.transparent {
  background: transparent;
}

.heading {
  margin: 0 0 utils.rem(spacing.$large);
}

.content {
  @include media-queries.mq($from: md) {
    display: flex;
  }
}

.media {
  margin-bottom: utils.rem(spacing.$large);

  @include media-queries.mq($from: md) {
    flex: 0 1 45%;
    margin: 0 utils.rem(spacing.$large) 0 0;
  }
}

.figure {
  margin: 0;
}

.caption {
  padding-top: utils.rem(spacing.$medium);
}

.storyList {
  list-style: none;
  margin: 0;
  max-width: none;
}

.stories {
  @include media-queries.mq($from: md) {
    flex: 0 1 55%;
  }
}
