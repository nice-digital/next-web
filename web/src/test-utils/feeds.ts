import fs from "fs/promises";
import path from "path";

import MockAdapter from "axios-mock-adapter";

import { client } from "../feeds";

const mockDataBaseDirectory = path.resolve(
		__dirname,
		"../",
		"__mocks__",
		"__data__"
	),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	publicationsBaseUrl = new URL(process.env.SERVER_FEEDS_PUBLICATIONS_ORIGIN!),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	inDevBaseUrl = new URL(process.env.SERVER_FEEDS_INDEV_ORIGIN!);

export const axiosJSONMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

/**
 * Adds default JSON endpoint mocks for publications and indev feeds.
 *
 * Maps the incoming request URL to static JSON files on the file system in the mock data folder
 */
export const addDefaultJSONFeedMocks = (): MockAdapter =>
	axiosJSONMock.onGet().reply(async (config) => {
		if (!config.url) throw Error(`URL was empty in Axios mock`);

		const { host, pathname } = new URL(config.url),
			isHTML = config.headers?.Accept === "text/html";

		let filePath = mockDataBaseDirectory;

		if (host === publicationsBaseUrl.host) {
			filePath = path.join(filePath, "publications");
		} else if (host === inDevBaseUrl.host) {
			filePath = path.join(filePath, "inDev");
		} else
			throw Error(
				`Unknown URL host for mocking. Expected either ${publicationsBaseUrl.host} or ${inDevBaseUrl.host} but got ${host}`
			);

		filePath = path.join(
			filePath,
			pathname.toLowerCase() + (isHTML ? ".html" : ".json")
		);

		try {
			await fs.access(filePath);
		} catch {
			throw Error(`Could not find mock JSON file with path ${filePath}`);
		}

		const fileContents = await fs.readFile(filePath, "utf8");

		return [
			200,
			isHTML ? fileContents : !fileContents ? "" : JSON.parse(fileContents),
		];
	});
