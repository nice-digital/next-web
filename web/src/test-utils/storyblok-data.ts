// Dump any useful test objects in here for mocking Storyblok responses

import { type StoryblokStory } from "storyblok-generate-ts";

import {
	type AuthorStoryblok,
	type RichtextStoryblok,
	type BlogPostStoryblok,
	type NewsArticleStoryblok,
	AssetStoryblok,
	PodcastStoryblok,
	InDepthArticleStoryblok,
} from "@/types/storyblok";

export const mockStoryblokStory = {
	id: 123456789,
	uuid: "abc123",
	name: "Title of the story",
	created_at: "1980-08-27T11:12:13.194Z",
	published_at: "2024-01-31T15:16:15.194Z",
	first_published_at: "2024-01-30T15:16:15.194Z",
	slug: "i-am-a-slug",
	full_slug: "i/am/a-very-full-slug",
	sort_by_date: null,
	position: 3,
	tag_list: ["rhubarb", "custard"],
	is_startpage: false,
	parent_id: 987654321,
	meta_data: null,
	group_id: "5625b207-7bb8-4410-aac3-3d7d5743f97a",
	release_id: null,
	lang: "default",
	path: undefined,
	alternates: [],
	default_full_slug: "default-full-slug",
	translated_slugs: [
		{
			path: "/translated-slug-path",
			name: "Translated slug name",
			lang: "en",
		},
	],
};

export const mockAuthorBlok: StoryblokStory<AuthorStoryblok> = {
	...mockStoryblokStory,
	name: "Test Author title",
	content: {
		component: "author",
		name: "Bob Holness",
		jobTitle: "Blockbusters host",
		image: {
			id: 123,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "Bob",
			alt: "Bob the host",
		},
		_uid: "123",
	},
};

export const mockRichText: RichtextStoryblok = {
	type: "doc",
	content: [
		{
			type: "heading",
			attrs: {
				level: 2,
			},
			content: [
				{
					text: "A mock h2 heading in a rich text field",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 3,
			},
			content: [
				{
					text: "A mock h3 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 4,
			},
			content: [
				{
					text: "A mock h4 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 5,
			},
			content: [
				{
					text: "A mock h5 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 6,
			},
			content: [
				{
					text: "A mock h6 heading",
					type: "text",
				},
			],
		},
		{
			type: "paragraph",
			content: [
				{
					type: "text",
					text: "Example text content.",
				},
			],
		},
		{
			type: "paragraph",
			content: [
				{
					type: "image",
					attrs: {
						id: "13894387",
						alt: "Kitten wearing mittens",
						src: "http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
						title: "",
						source: "",
						copyright: "",
						meta_data: {},
					},
				},
			],
		},
		{
			type: "blockquote",
			content: [
				{
					type: "paragraph",
					content: [
						{
							text: "A inline quote in the rich text field of a news article. ",
							type: "text",
						},
					],
				},
			],
		},
	],
};

export const mockNewsArticle: StoryblokStory<NewsArticleStoryblok> = {
	...mockStoryblokStory,
	id: 987654,
	name: "Test news article title",
	content: {
		title: "Test news article title",
		date: "2024-01-31T15:16:15.19",
		introText: "Test intro text",
		image: {
			id: 123,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "A kitten being a kitten",
			alt: "Kitten mittens",
		},
		_uid: "abc123",
		component: "newsArticle",
		content: { ...mockRichText },
		resources: [
			{
				_uid: "8f019cf9-3743-4126-94cb-092898b43636",
				link: {
					id: "e9c6fedf-5d63-4468-b29f-8884d5300ad8",
					url: "",
					linktype: "story",
					fieldtype: "multilink",
					cached_url: "test-new-nc-2",
				},
				title: "Test related link 1",
				component: "relatedLink",
				_editable:
					'<!--#storyblok#{"name": "relatedLink", "space": "271255", "uid": "8f019cf9-3743-4126-94cb-092898b43636", "id": "435600482"}-->',
			},
		],
		relatedNews: [
			{
				_uid: "50c7601a-1f1f-48f1-96e9-4f16f688fd7a",
				date: "2024-01-31 00:00",
				link: {
					id: "",
					url: "https://youtube.com/@niceorguk",
					linktype: "url",
					fieldtype: "multilink",
					cached_url: "https://youtube.com/@niceorguk",
				},
				title: "Test related news link 1",
				component: "relatedNewsLink",
				publisher: "NICE",
				_editable:
					'<!--#storyblok#{"name": "relatedNewsLink", "space": "271255", "uid": "50c7601a-1f1f-48f1-96e9-4f16f688fd7a", "id": "435600482"}-->',
			},
		],
	},
};

export const mockBlogPost: StoryblokStory<BlogPostStoryblok> = {
	...mockStoryblokStory,
	name: "Test blog post title",
	content: {
		title: "Test blog post title",
		date: "2024-02-20",
		introText: "Let's play Blockbusters!",
		author: [{ ...mockAuthorBlok }],
		image: {
			id: 456,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "Cat cat cat",
			alt: "A kitty cat",
		},
		_uid: "def456",
		component: "blogPost",
		content: { ...mockRichText },
	},
};

export const mockPodcastPage: StoryblokStory<PodcastStoryblok> = {
	...mockStoryblokStory,
	name: "Test podcast title",
	content: {
		_uid: "2663bc46-cefe-4878-8c86-ecc78cc1f65e",
		date: "2024-04-17 12:00",
		image: {
			id: 15267557,
			alt: "",
			name: "",
			focus: "",
			title: "",
			source: "",
			filename:
				"https://local-host-test-nice-org.com/f/243782/413x297/6a7748fdb4/stroke.PNG",
			copyright: "",
			fieldtype: "asset",
			meta_data: {},
			is_private: false,
			is_external_url: false,
		},
		component: "podcast",
		introText:
			"It is estimated that there are around 85,000 strokes every year in England, with over 1 million people currently living with its long-term effects.\n" +
			"\n" +
			"We catch up with Jon Brown, chair of the independent committee and Professor Stephen Hill, lay member on the committee about NICE’s updated guidance on stroke rehabilitation.",
		description: {
			type: "doc",
			content: [
				{
					type: "paragraph",
					content: [
						{
							text: "It is estimated that there are around 85,000 strokes every year in England, with over 1 million people currently living with its long-term effects.",
							type: "text",
						},
					],
				},
			],
		},
		soundcloudEmbedID: "1648857267",
	},
};

export const mockImageAsset: AssetStoryblok = {
	filename:
		"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
	id: 654654654,
	name: "Delicious bacon",
};

export const mockIndepthArticle: StoryblokStory<InDepthArticleStoryblok> = {
	...mockStoryblokStory,
	name: "sample indepth test",
	id: 476215653,
	uuid: "7bb356cb",
	content: {
		_uid: "cd984458-0e14-4fd6-871d-469e9cd3a845",
		date: "2024-04-10 12:00",
		link: {
			id: "",
			url: "https://local-host-test-nice-org.com/news/blogs/lessons-learned-from-the-multi-agency-advisory-service",
			linktype: "url",
			fieldtype: "multilink",
			cached_url:
				"https://local-host-test-nice-org.com/news/blogs/lessons-learned-from-the-multi-agency-advisory-service",
		},
		image: mockImageAsset,
		title: "sample test",
		component: "inDepthArticle",
		introText:
			"Clíodhna Ní Ghuidhir and Rebecca Boffa outline the common challenges that developers of artificial intelligence (AI) tech face when navigating their way to the health and care market, and link to their series of articles, which provide further advice on what can be done to avoid these.",
	},
};
