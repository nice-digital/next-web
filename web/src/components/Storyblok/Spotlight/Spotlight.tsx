import { ISbStoryData } from "@storyblok/react";
import classnames from "classnames";

import { Card, type CardHeadingLinkProps } from "@nice-digital/nds-card";
import { Tag } from "@nice-digital/nds-tag";

import { Link } from "@/components/Link/Link";
import { StoryblokImage } from "@/components/Storyblok/StoryblokImage/StoryblokImage";
import { StoryblokYoutubeEmbed } from "@/components/Storyblok/StoryblokYoutubeEmbed/StoryblokYoutubeEmbed";
import { NewsStory } from "@/types/News";
import { SpotlightStoryblok, YoutubeEmbedStoryblok } from "@/types/storyblok";
import { defaultPodcastImage, getNewsType, newsTypes } from "@/utils/storyblok";

import styles from "./Spotlight.module.scss";

export interface SpotlightProps {
	blok: SpotlightStoryblok;
	headingLevel?: number;
	className?: string;
}

export const Spotlight: React.FC<SpotlightProps> = ({
	blok,
	headingLevel = 2,
	className,
}: SpotlightProps) => {
	const { heading, mediaDescription, youtubeEmbed, isTransparent } = blok;

	// Resolve heading type
	const HeadingElement = `h${headingLevel}` as keyof JSX.IntrinsicElements;

	// Resolve stories
	const stories = blok.stories as ISbStoryData<NewsStory>[];

	// Resolve transparency
	const transparentClass = isTransparent ? styles.transparent : undefined;
	const cardHeadingLevel =
		headingLevel && headingLevel < 6
			? (headingLevel + 1).toString()
			: undefined;
	return (
		<section
			className={classnames(styles.spotlight, transparentClass, className)}
			data-tracking="spotlight"
		>
			<div className={styles.container}>
				<HeadingElement className={styles.heading}>{heading}</HeadingElement>
				<div className={styles.content}>
					<div className={styles.media}>
						<figure className={styles.figure}>
							<StoryblokYoutubeEmbed
								blok={youtubeEmbed[0] as YoutubeEmbedStoryblok}
							/>
							{mediaDescription && (
								<figcaption className={styles.caption}>
									{mediaDescription}
								</figcaption>
							)}
						</figure>
					</div>
					<div className={styles.stories}>
						<ul className={styles.storyList}>
							{stories.map((story) => {
								const { content, full_slug, id, name } = story;

								// Resolve story type
								const storyType = getNewsType(content.component);

								const headingLink: CardHeadingLinkProps =
									storyType === newsTypes.inDepthArticle
										? {
												destination:
													content.link.url || content.link.cached_url,
												elementType: "a",
										  }
										: {
												destination: full_slug,
												elementType: Link,
										  };

								// Fall back to podcast placeholder image if none is supplied
								const image = content.image?.filename || defaultPodcastImage;

								return (
									<Card
										key={id}
										headingText={name}
										headingLevel={cardHeadingLevel}
										elementType="li"
										link={headingLink}
										image={
											<StoryblokImage
												src={image}
												alt={content.image?.alt || ""}
												serviceOptions={{ height: 0, quality: 80, width: 708 }}
											/>
										}
									>
										<Tag outline>{storyType}</Tag>
									</Card>
								);
							})}
						</ul>
					</div>
				</div>
			</div>
		</section>
	);
};
