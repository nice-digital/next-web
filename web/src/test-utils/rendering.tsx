/* eslint-disable import/export, import/no-unresolved -- We're deliberately overrind the default render from @testing-library with a custom render method */
import {
	render as testingLibrRender,
	RenderResult,
} from "@testing-library/react";
import { HeadManagerContext } from "next/dist/shared/lib/head-manager-context.shared-runtime";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { NextRouter } from "next/router";
import { ReactElement, useEffect } from "react";
import { renderToString } from "react-dom/server";

import NextWebApp from "../pages/_app";

import type { AppProps } from "next/app";

/**
 *
 */
export const mockRouter: NextRouter = {
	basePath: "",
	pathname: "/",
	route: "/",
	asPath: "/",
	query: {},
	push: jest.fn(),
	replace: jest.fn(),
	reload: jest.fn(),
	back: jest.fn(),
	forward: jest.fn(),
	prefetch: jest.fn(),
	beforePopState: jest.fn(),
	events: {
		on: jest.fn(),
		off: jest.fn(),
		emit: jest.fn(),
	},
	isFallback: false,
	isLocaleDomain: false,
	isPreview: false,
	isReady: false,
};

/**
 * Convenience method for getting a mock router with the given properties.
 *
 * @example
 * 	getMockRouter({ pathname: "/about" })
 *
 * @param routerParams Optional router properties that get merged with the default mock router
 * @returns A new mock router object
 */
export const getMockRouter = (
	routerParams?: Partial<NextRouter>
): NextRouter => ({
	...mockRouter,
	...routerParams,
});

/**
 * Collect tags from next/head: https://edibleco.de/3Anu92S
 */
const HeadProvider: React.FC = ({ children }) => {
	let head: JSX.Element[];

	useEffect(() => {
		global.document.head.insertAdjacentHTML(
			"afterbegin",
			renderToString(<>{head}</>) || ""
		);
	});

	return (
		<HeadManagerContext.Provider
			value={{
				updateHead: (state) => {
					head = state;
				},
				mountedInstances: new Set(),
			}}
		>
			{children}
		</HeadManagerContext.Provider>
	);
};

export type GetWrapperOptions = {
	router: NextRouter;
	wrapper: RenderOptions["wrapper"];
};
const getAppWrapper =
	({ router, wrapper: Wrapper }: GetWrapperOptions) =>
	({ children }: { children: ReactElement }) =>
		(
			<HeadProvider>
				<RouterContext.Provider value={router}>
					<NextWebApp
						Component={() => (
							<>{Wrapper ? <Wrapper>{children}</Wrapper> : children}</>
						)}
						pageProps={{}}
						router={router as AppProps["router"]}
					/>
				</RouterContext.Provider>
			</HeadProvider>
		);

export type RenderOptions = Parameters<typeof testingLibrRender>[1] & {
	router?: Partial<NextRouter>;
};

/**
 * Custom render method the given component, wrapped in `NextWebApp` and a `HeadProvider`.
 *
 * @see https://testing-library.com/docs/react-testing-library/setup/#custom-render
 *
 * @param ui The element to render
 * @param options Optional render options that get passed to underlying React Testing Lib render method
 * @returns The render result
 */
const customRender = (
	ui: ReactElement,
	{ wrapper, router, ...options }: RenderOptions = {}
): RenderResult =>
	testingLibrRender(ui, {
		wrapper: getAppWrapper({ router: getMockRouter(router), wrapper }),
		...options,
	});

export * from "@testing-library/react";
export { customRender as render };
