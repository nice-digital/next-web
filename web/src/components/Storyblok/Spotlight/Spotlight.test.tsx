import { render, screen, within } from "@testing-library/react";

import { mockNewsArticle, mockBlogPost } from "@/test-utils/storyblok-data";

import { Spotlight, type SpotlightProps } from "./Spotlight";

const mockSpotlight: SpotlightProps = {
	blok: {
		heading: "Mock spotlight title",
		mediaDescription: "Mock media description",
		_uid: "123456789",
		component: "spotlight",
		youtubeEmbed: [
			{
				title: "Test Youtube title",
				source: "wwSzpaTHyS8",
				_uid: "123456",
				component: "youtubeEmbed",
			},
		],
		stories: [mockNewsArticle, mockBlogPost],
	},
};

describe("Spotlight component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Spotlight {...mockSpotlight} />);
		expect(container).toMatchSnapshot();
	});

	it("should render a heading at the correct level if one is specified", () => {
		const spotlight: SpotlightProps = {
			...mockSpotlight,
			headingLevel: 5,
		};

		render(<Spotlight {...spotlight} />);
		expect(
			screen.getByRole("heading", { level: 5, name: "Mock spotlight title" })
		).toBeInTheDocument();
	});
	it("should render each Card with a heading level one deeper than the Spotlight heading", () => {
		const headingLevel = 3; // So Card headings should be level 4
		render(<Spotlight {...mockSpotlight} headingLevel={headingLevel} />);

		const cards = screen.getAllByRole("listitem");
		for (const card of cards) {
			const heading = within(card).getByRole("heading", {
				level: headingLevel + 1,
			});
			expect(heading).toBeInTheDocument();
		}
	});

	it("should not exceed heading level 6 for cards", () => {
		const headingLevel = 6; // cardHeadingLevel becomes undefined
		render(<Spotlight {...mockSpotlight} headingLevel={headingLevel} />);

		const articleLink = screen.getByText("Test news article title");
		expect(articleLink).toBeInTheDocument();
		expect(articleLink.parentElement?.tagName).toBe("P");

		const blogLink = screen.getByText("Test blog post title");
		expect(blogLink).toBeInTheDocument();
		expect(blogLink.parentElement?.tagName).toBe("P");
	});
});
