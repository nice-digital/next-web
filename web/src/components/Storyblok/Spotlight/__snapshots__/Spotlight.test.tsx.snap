// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotlight component should match snapshot 1`] = `
<div>
  <section
    class="spotlight"
    data-tracking="spotlight"
  >
    <div
      class="container"
    >
      <h2
        class="heading"
      >
        Mock spotlight title
      </h2>
      <div
        class="content"
      >
        <div
          class="media"
        >
          <figure
            class="figure"
          >
            <link
              as="image"
              href="https://i.ytimg.com/vi_webp/wwSzpaTHyS8/hqdefault.webp"
              rel="prefetch"
            />
            <article
              class="youtubeWrapper "
              data-title="(video) Test Youtube title"
              style="background-image: url(https://i.ytimg.com/vi_webp/wwSzpaTHyS8/hqdefault.webp); --aspect-ratio: 56.25%;"
            >
              <button
                aria-label="Watch (video) Test Youtube title"
                class="youtubePlayButton"
                type="button"
              />
            </article>
            <figcaption
              class="caption"
            >
              Mock media description
            </figcaption>
          </figure>
        </div>
        <div
          class="stories"
        >
          <ul
            class="storyList"
          >
            <li
              class="card"
              data-component="card"
            >
              <div
                class="card__image"
              >
                <picture>
                  <source
                    srcset="http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29/m/708x0/filters:format%28avif%29:quality%2880%29"
                    type="image/avif"
                  />
                  <source
                    srcset="http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29/m/708x0/filters:format%28webp%29:quality%2880%29"
                    type="image/webp"
                  />
                  <img
                    alt="Kitten mittens"
                    data-nimg="1"
                    decoding="async"
                    height="0"
                    loading="lazy"
                    src="/_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=1920&q=75"
                    srcset="/_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 1x, /_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=1920&q=75 2x"
                    style="color: transparent;"
                    width="708"
                  />
                </picture>
              </div>
              <div
                class="card__text"
              >
                <header
                  class="card__header"
                >
                  <p
                    class="card__heading"
                  >
                    <a
                      href="i/am/a-very-full-slug"
                    >
                      Test news article title
                    </a>
                  </p>
                </header>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
              </div>
            </li>
            <li
              class="card"
              data-component="card"
            >
              <div
                class="card__image"
              >
                <picture>
                  <source
                    srcset="http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29/m/708x0/filters:format%28avif%29:quality%2880%29"
                    type="image/avif"
                  />
                  <source
                    srcset="http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29/m/708x0/filters:format%28webp%29:quality%2880%29"
                    type="image/webp"
                  />
                  <img
                    alt="A kitty cat"
                    data-nimg="1"
                    decoding="async"
                    height="0"
                    loading="lazy"
                    src="/_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=1920&q=75"
                    srcset="/_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 1x, /_next/image?url=http%3A%2F%2Flocal-host-test-nice-org.com%2Ff%2F243782%2F420x234%2F7a576133e7%2Fregister-for-board-meeting.PNG%2Fm%2F867x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529%2Fm%2F708x0%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=1920&q=75 2x"
                    style="color: transparent;"
                    width="708"
                  />
                </picture>
              </div>
              <div
                class="card__text"
              >
                <header
                  class="card__header"
                >
                  <p
                    class="card__heading"
                  >
                    <a
                      href="i/am/a-very-full-slug"
                    >
                      Test blog post title
                    </a>
                  </p>
                </header>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  Blogs
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</div>
`;
